extensions/data-transfer/portability-data-transfer-google/src/test/java/org/dataportabilityproject/datatransfer/google/mail/GoogleMailExporterTest.java

```diff
@@ -16,6 +16,162 @@
+import org.dataportabilityproject.types.transfer.models.mail.MailContainerResource;
+import org.dataportabilityproject.types.transfer.models.mail.MailMessageModel;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.InOrder;
+import org.mockito.Matchers;
+import org.mockito.Mockito;
+
 public class GoogleMailExporterTest {
 
+  private static final UUID JOB_ID = UUID.randomUUID();
+
+  private static final String NEXT_TOKEN = "next_token";
+
+  private static final String MESSAGE_ID = "messageId";
+  private static final String MESSAGE_RAW = "message contents";
+  private static final List<String> MESSAGE_LABELS = ImmutableList.of("label1", "label2");
+  private static final Message INITIAL_MESSAGE = new Message().setId(MESSAGE_ID);
+  private static final Message FULL_MESSAGE = new Message().setId(MESSAGE_ID).setRaw(MESSAGE_RAW)
+      .setLabelIds(MESSAGE_LABELS);
+
+  private Users users;
+  private Messages messages;
+  private Messages.List messageListRequest;
+  private ListMessagesResponse messageListResponse;
+  private Get get;
+
+  private Gmail gmail;
+
+  private GoogleMailExporter googleMailExporter;
+
+  @Before
+  public void setup() throws IOException {
+    users = mock(Users.class);
+    messages = mock(Messages.class);
+    messageListRequest = mock(Messages.List.class);
+    get = mock(Get.class);
+    gmail = mock(Gmail.class);
+
+    GoogleCredentialFactory googleCredentialFactory = mock(GoogleCredentialFactory.class);
+    googleMailExporter = new GoogleMailExporter(googleCredentialFactory, gmail);
+
+    when(gmail.users()).thenReturn(users);
+    when(users.messages()).thenReturn(messages);
+    when(messages.list(Matchers.anyString())).thenReturn(messageListRequest);
+    when(messageListRequest.setMaxResults(Matchers.anyLong())).thenReturn(messageListRequest);
+    when(messages.get(Matchers.anyString(), Matchers.anyString())).thenReturn(get);
+    when(get.setFormat(Matchers.anyString())).thenReturn(get);
+
+    verifyZeroInteractions(googleCredentialFactory);
+  }
+
+  @Test
+  public void exportMessagesFirstSet() throws IOException {
+    setUpSingleMessageResponse();
+
+    // Looking at first page, with at least one page after it
+    messageListResponse.setNextPageToken(NEXT_TOKEN);
+
+    // Run test
+    ExportResult<MailContainerResource> result =
+        googleMailExporter.export(JOB_ID, null, Optional.empty());
+
+    // Check results
+    // Verify correct methods were called
+    InOrder inOrder = Mockito.inOrder(gmail, users, messages, messageListRequest, get);
+    // First request
+    inOrder.verify(gmail).users();
+    inOrder.verify(users).messages();
+    inOrder.verify(messages).list(GoogleMailExporter.USER);
+    inOrder.verify(messageListRequest).setMaxResults(GoogleMailExporter.PAGE_SIZE);
+    inOrder.verify(messageListRequest).execute();
+    verify(messageListRequest, never()).setPageToken(Matchers.anyString());
+    // Second request
+    inOrder.verify(gmail).users();
```

Comment:
Same here, can we simply this group of verifications without having to call each one?